diff -u -r glibc-2.3.5/ctype/ctype-info.c glibc-2.3.5.new/ctype/ctype-info.c
--- glibc-2.3.5/ctype/ctype-info.c	2005-10-18 22:04:44.000000000 +1000
+++ glibc-2.3.5.new/ctype/ctype-info.c	2005-10-18 22:05:03.000000000 +1000
@@ -48,18 +48,18 @@
 
 #define b(t,x,o) (((const t *) _nl_C_LC_CTYPE_##x) + o)
 
-const unsigned short int *__ctype_b = b (unsigned short int, class, 128);
-const __uint32_t *__ctype32_b = b (__uint32_t, class32, 0);
-const __int32_t *__ctype_tolower = b (__int32_t, tolower, 128);
-const __int32_t *__ctype_toupper = b (__int32_t, toupper, 128);
-const __uint32_t *__ctype32_tolower = b (__uint32_t, tolower, 128);
-const __uint32_t *__ctype32_toupper = b (__uint32_t, toupper, 128);
+const unsigned short int *__old_ctype_b = b (unsigned short int, class, 128);
+const __uint32_t *__old_ctype32_b = b (__uint32_t, class32, 0);
+const __int32_t *__old_ctype_tolower = b (__int32_t, tolower, 128);
+const __int32_t *__old_ctype_toupper = b (__int32_t, toupper, 128);
+const __uint32_t *__old_ctype32_tolower = b (__uint32_t, tolower, 128);
+const __uint32_t *__old_ctype32_toupper = b (__uint32_t, toupper, 128);
 
-compat_symbol (libc, __ctype_b, __ctype_b, GLIBC_2_0);
-compat_symbol (libc, __ctype_tolower, __ctype_tolower, GLIBC_2_0);
-compat_symbol (libc, __ctype_toupper, __ctype_toupper, GLIBC_2_0);
-compat_symbol (libc, __ctype32_b, __ctype32_b, GLIBC_2_0);
-compat_symbol (libc, __ctype32_tolower, __ctype32_tolower, GLIBC_2_2);
-compat_symbol (libc, __ctype32_toupper, __ctype32_toupper, GLIBC_2_2);
+compat_symbol (libc, __old_ctype_b, __ctype_b, GLIBC_2_0);
+compat_symbol (libc, __old_ctype_tolower, __ctype_tolower, GLIBC_2_0);
+compat_symbol (libc, __old_ctype_toupper, __ctype_toupper, GLIBC_2_0);
+compat_symbol (libc, __old_ctype32_b, __ctype32_b, GLIBC_2_0);
+compat_symbol (libc, __old_ctype32_tolower, __ctype32_tolower, GLIBC_2_2);
+compat_symbol (libc, __old_ctype32_toupper, __ctype32_toupper, GLIBC_2_2);
 
 #endif
